docs/best_practices/nwbfile_metadata.rst

```diff
@@ -1,83 +1,192 @@
+any other environmental sensors.
+
+The 'processing' modules are for intermediate data, *e.g.*, if you derive the position of the animal from sensors or
+video tracking, that would go in ``/processing/behavior/Position/SpatialSeries``. An LFP signal that is a downsampled
+version of the acquired data would go in ``/processing/ecephys/LFP/ElectricalSeries``.
+
+It may not always be 100% clear whether data is acquired or derived, so in those cases you should just use your best
+judgement.
+
+
+
+.. _best_practice_processing_module_name:
+
+Processing Module Names
+~~~~~~~~~~~~~~~~~~~~~~~
 
-The 'processing' modules are for intermediate data. If you take the wheel position and derive the position of the animal
-in meters, that would go in ```/processing/behavior/Position/SpatialSeries```.
+The name of any given processing module should be chosen from the following types: "ophys", "ecephys", "icephys",
+"behavior", "ogen", "retinotopy", or "misc". This helps standardize navigation of NWBFiles generated from labs and
+modalities. It also helps distinguish components of a given experiment, such as decoupling the intermediate data from
+neural acquisition systems from behavioral ones.
 
-An LFP signal that is a downsampled version of the acquired data would go in ```/processing/ecephys/LFP/ElectricalSeries```.
+Check function: :py:meth:`~nwbinspector.checks.nwbfile_metadata.check_processing_module_name`
 
-It may not always be 100% clear whether data is acquired or derived, so in those cases you should just use your best judgement.
 
 
+File Metadata
+-------------
+
+
+.. _best_practice_file_id:
+
 File Identifiers
 ~~~~~~~~~~~~~~~~
 
-NWBFile has two distinct places for ids: session_id and identifier.
+An :nwb-schema:ref:`sec-NWBFile` has two distinct places for identifiers: ``NWBFile.session_id`` and ``identifier``.
+
+The ``session_id`` field marks unique experimental sessions. The ``session_id`` should have a one-to-one relationship
+with a recording session. Sometimes you may find yourself having multiple NWBFiles that correspond to the same session.
+This can happen, for instance, if you separate out processing steps across multiple files or if you want to compare
+different processing outputs. In this case, the ``session_id`` should be the same for each file. Each lab should follow
+a standard structure for their own naming schemes so that sessions are unique within the lab and the IDs are easily
+human-readable.
+
+The ``identifier`` tag should be a globally unique value for the :nwb-schema:ref:`sec-NWBFile`. Two different NWBFiles
+from the same session should have different ``identifier`` values if they differ in any way. It is recommended that you
+use a well-established algorithmic generator such as :uuid:`uuid <>` (for PyNWB) or ?? (for MatNWB) to ensure
+uniqueness. The ``identifier`` does not need to be easily human-readable.
```

Comment:
I noticed MATLAB does have a `uuid` package of some sort, but I couldn't find example usage via MatNWB documentation; what should our recommendation be?